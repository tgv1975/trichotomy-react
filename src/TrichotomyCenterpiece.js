import React, { Component } from 'react';
import PropTypes from 'prop-types';

import trichotomy from 'trichotomy/dist/trichotomy.min.css';

export default class TrichotomyCenterpiece extends Component {

    static propTypes = {

    }

    render() {
        const {
            children,
            className
        } = this.props;

        let cssClass = (className &&
            className.concat(' ', trichotomy.centerpiece)) ||
            trichotomy.centerpiece;

        return (
            <div className={cssClass}>
                {children}
            </div>
        );
    }
}
