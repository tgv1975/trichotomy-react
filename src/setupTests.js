jest.mock('trichotomy/dist/trichotomy.min.css', () => (
    {
        trichotomy: 'trichotomy',
        vertical: 'vertical',
        horizontal: 'horizontal',
        right: 'right',
        centerpiece: 'centerpiece',
    }
));
