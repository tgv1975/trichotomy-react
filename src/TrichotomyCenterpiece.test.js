import React from 'react';
import renderer from 'react-test-renderer';
import './setupTests';

import TrichotomyCenterpiece from './TrichotomyCenterpiece';

describe('Trichotomy', () => {
    it('Trichotomy is truthy', () => {
        expect(TrichotomyCenterpiece).toBeTruthy();
    });

    test('Renders TrichotomyCenterpiece container <DIV> element (default)', () => {
        const component = renderer.create(
            <TrichotomyCenterpiece />,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.type).toEqual('div');
    });

    test('Renders TrichotomyCenterpiece container <ARTICLE> element (custom)', () => {
        const component = renderer.create(
            <TrichotomyCenterpiece tag="article" />,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.type).toEqual('article');
    });

    test('Renders TrichotomyCenterpiece container element with "centerpiece" CSS class', () => {
        const component = renderer.create(
            <TrichotomyCenterpiece><div></div></TrichotomyCenterpiece>,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.props.className).toEqual('centerpiece');
    });

    test('Renders TrichotomyCenterpiece container element with "someCustomClass centerpiece" CSS class if set', () => {
        const component = renderer.create(
            <TrichotomyCenterpiece className="someCustomClass" />,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.props.className).toEqual('someCustomClass centerpiece');
    });

    test('Renders children into TrichotomyCenterpiece container element ', () => {
        const component = renderer.create(
            <TrichotomyCenterpiece direction="vertical">
                <div>First child</div>
                <div>Second child</div>
                <div>Third child</div>
            </TrichotomyCenterpiece>,
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
        expect(tree.children).toBeTruthy();
    });
});
