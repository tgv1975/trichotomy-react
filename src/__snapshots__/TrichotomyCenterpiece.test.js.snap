// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trichotomy Renders TrichotomyCenterpiece container <ARTICLE> element (custom) 1`] = `
<article
  className="centerpiece"
/>
`;

exports[`Trichotomy Renders TrichotomyCenterpiece container <DIV> element (default) 1`] = `
<div
  className="centerpiece"
/>
`;

exports[`Trichotomy Renders TrichotomyCenterpiece container element with "centerpiece" CSS class 1`] = `
<div
  className="centerpiece"
>
  <div />
</div>
`;

exports[`Trichotomy Renders TrichotomyCenterpiece container element with "someCustomClass centerpiece" CSS class if set 1`] = `
<div
  className="someCustomClass centerpiece"
/>
`;

exports[`Trichotomy Renders children into TrichotomyCenterpiece container element  1`] = `
<div
  className="centerpiece"
>
  <div>
    First child
  </div>
  <div>
    Second child
  </div>
  <div>
    Third child
  </div>
</div>
`;
